src/components/overlays/quiz-overlay.js

```diff
@@ -0,0 +1,379 @@
+          color: inherit;
+          font-weight: bold;
+        }
+
+        .quiz__highlighting-none {
+          color: inherit;
+        }
+
+        .quiz__highlighting-correct {
+          color: lime;
+        }
+
+        .quiz__highlighting-wrong {
+          color: red;
+        }
+
+        .quiz__highlighting-strikethrough label {
+          text-decoration: line-through;
+        }
+
+        #input__freetext-answer {
+          background: transparent;
+          border: none;
+          border-bottom: 1px dashed;
+          text-align: center;
+          width: 300px;
+          color: inherit;
+        }
+      </style>
+
+      <div id="container__quiz-overlay" class="overlay" style$="visibility: [[ifThenElse(_isVisible, 'visible', 'hidden')]];">
+        <h3 id="text__quiz-question">[[_currentQuestion.text]]</h3>
+
+        <template is="dom-if" if="[[_isTextQuestion(_currentQuestion)]]">
+          <div class$="[[_getValidationClass(_currentQuestion, _correctAnswersShown, _correctAnswers, _isAnswerCorrect, null)]]">
+            <input id="input__freetext-answer" type="text" on-input="_textAnswerChanged" placeholder$="[[localize('quiz--enter-answer-here')]]" disabled$="[[_correctAnswersShown]]"></input>
+          </div>
+
+          <template is="dom-if" if="[[_correctAnswersShown]]">
+            <h5>[[localize('quiz--correct-answers')]]</h5>
+            <div class="container__answer-box">
+              <template is="dom-repeat" items="[[_correctAnswers]]">
+                <span class="">[[item.text]]</span>
+              </template>
+            </div>
+          </template>
+        </template>
+
+        <template is="dom-if" if="[[!_isTextQuestion(_currentQuestion)]]">
+          <div class="container__answer-box">
+            <template is="dom-repeat" items="[[_currentQuestion.answers]]">
+              <div class$="[[_getValidationClass(_currentQuestion, _correctAnswersShown, _correctAnswers, _isAnswerCorrect, item)]]">
+                <div class="form-check">
+                  <input name="answer" class="select__quiz-answer form-check-input" value$="[[item.id]]" id="select__quiz-answer-[[item.id]]" type$="[[ifThenElse(_isSingleChoiceQuestion, 'radio', 'checkbox')]]" disabled$="[[_correctAnswersShown]]" on-change="_selectionChanged"></input>
+                  <label class="form-check-label" for$="select__quiz-answer-[[item.id]]">[[item.text]]</label>
+                </div>
+              </div>
+            </template>
+          </div>
+        </template>
+
+        <div class="container__centered-row">
+          <template is="dom-if" if="[[_correctAnswersShown]]">
+            <a class="button button__quiz" on-click="_handleSkipClick" href="#">[[localize('general--continue')]]</a>
+          </template>
+
+          <template is="dom-if" if="[[!_correctAnswersShown]]">
+            <a class="button button__quiz" on-click="_handleSkipClick" href="#">[[localize('quiz--skip-question')]]</a>
```

Comment:
The onclick functions call event.preventDefault, so the default click action will not trigger. I intentionally did it this way (href="#", preventDefault in javascript) because for me as someone who uses script-blocking extensions a lot, whenever I click a button and the page just jumps to the top, I know instantly that it's due to a missing script. If I click it and nothing happens, you don't really know where it's broken. We can change it to javascript:void(0) though, if you like.

Reply:
When JavaScript is disabled, the player is not rendered anyway. Therefore, I prefer removing this behavior here. I think you can even omit the `href` attribute at all.

Reply:
check